_inc/client/plans/single-product-backup.jsx

```diff
@@ -155,15 +186,93 @@ function PlanRadioButton( {
 	);
 }
 
-// eslint-disable-next-line no-unused-vars
+function PlanSavingsText( { selectedBackup, billingTimeFrame, currencyCode } ) {
+	if ( ! selectedBackup ) {
+		return null;
+	}
+	const savingsCurrencyObject = getCurrencyObject( selectedBackup.potentialSavings, currencyCode );
+	const savings = formatCurrency( savingsCurrencyObject );
+
+	// TODO: Provide way to switch billing time frame state.
+	const switchComponent = <a />;
+
+	let promptText;
+	if ( 'yearly' === billingTimeFrame ) {
+		promptText = __( 'You are saving %(savings)s by paying yearly. {{a}}Switch to monthly{{/a}}', {
+			args: { savings },
+			components: {
+				a: switchComponent,
+			},
+		} );
+	} else if ( 'monthly' === billingTimeFrame ) {
+		promptText = __(
+			'You could be saving %(savings)s by paying yearly. {{a}}Switch to yearly{{/a}}',
+			{
+				args: { savings },
+				components: {
+					a: switchComponent,
+				},
+			}
+		);
+	}
+
+	if ( ! promptText ) {
+		return null;
+	}
+	return (
+		<p>
+			<em>{ promptText }</em>
+		</p>
+	);
+}
+
+function UpgradeButton( { selectedUpgrade, billingTimeFrame, currencyCode, onClickHandler } ) {
+	if ( ! selectedUpgrade ) {
+		return null;
+	}
+	const { link, name, fullPrice, discountedPrice, type } = selectedUpgrade;
+	let billingTimeFrameString = '';
+	if ( 'yearly' === billingTimeFrame ) {
+		billingTimeFrameString = __( 'per year', { context: 'Amount of money per time unit.' } );
+	} else if ( 'monthly' === billingTimeFrame ) {
+		billingTimeFrameString = __( 'per month', { context: 'Amount of money per time unit.' } );
+	}
+
+	const currencyObject = getCurrencyObject( discountedPrice || fullPrice, currencyCode );
+
+	return (
+		<div className="single-product-backup__upgrade-button-container">
+			<Button href={ link } onClick={ onClickHandler( type ) } primary>
+				{ __( 'Upgrade to %(name)s for %(price)s/%(billingTime)s', {
+					args: {
+						name,
+						billingTime: billingTimeFrameString,
+						price: formatCurrency( currencyObject ),
```

Comment:
Could we, by any chance, reuse here the [`PlanPrice` component](https://github.com/Automattic/jetpack/blob/2461484dad9f4bfa8f963552f0a6235920939123/_inc/client/components/plans/plan-price/index.jsx)? We would have all the currency handling out-of-the box.

For sure one obstacle is the fact that the `PlanPrice` returns `h4` element and we want to use it here inside a `button`. It'd have to be modified to return just a `span`, for instance. Alternatively, it could accept an argument to tell if it should return a `h4` tag or just a `Fragment` instead, giving us a full flexibility.

What do you think?

Reply:
I think modifying the `PlanPrice` component would probably be the way to go, honestly. I didn't use it because of the `h4`, but you're right that adding a prop there would allow us to reuse it.

Reply:
I went ahead and issued a PR that adds an `inline` option to the `PlanPrice` component (and also refactors it): #14231 

Please review and feel free to re-use it here now or after it's merged (if the #14231 gets into master after this one).